Source/Tools/LinuxEmulation/LinuxSyscalls/ThreadManager.h

```diff
@@ -22,6 +22,79 @@ namespace FEX::HLE {
 class SyscallHandler;
 class SignalDelegator;
 
+// A latch that can be inspected to see if there is a waiter currently active.
+// This allows us to remove the race condition between a thread trying to go asleep and something else telling it to go to sleep or wake up.
```

Comment:
This race condition really comes out of nowhere, making the entire line very vague. I think by trying to be very specific (i.e. mentioning the concrete example), it actually gets in the way of conveying the point.

What you're really trying to say here is that entering waiting state and signaling aren't required to happen in any specific order:

```
In contrast to a condition variable, an InspectableLatch may be signaled *before* being waited on without change of effect.
```

Reply:
Sure, updated comment.